### PR TITLE
Enable passing of a length 2 vector as name argument (addresses #359)

### PR DESCRIPTION
**Pull request by francisbarton**
## Description
Allow the `name` argument to accept a vector, as well as a single value (as currently).

If a vector of length 2 is supplied, and `where` is equal to `c("row", "col")`, the first element will be used as the first column in the appended totals row (unless column 1 is included in cols to be totalled), and the second element will be used as the column name in the appended totals column.
This allows the user to give different headers to a totals row and a totals column, where previously these had to have the same value.

If a single item is supplied, this is duplicated into a vector of length 2. This will supply the same name for the row and column headers, as above - and providing the same functionality as previously - if both "row" and "col" are stipulated.

If only one of "row" or "col" is stipulated, it still works, as the function will just take either the first or the second element as necessary (and both are identical).

If only one of "row" or "col" is stipulated, but the user in error provides a longer vector for `name`, the vector is trimmed back leaving just the first element.

I have also included suggested revisions (commented out) to the add totals row and add totals column sections, to use a pure dplyr process. But it may be that these result in forcing tibble output rather than data.frame output, so please ignore and delete suggested lines if unwanted.

## Related Issue
Addresses feature request in #359

## Example
No change to existing usage; pure addition of extra option.

No tests included.

PR includes 4 previous commits which were just bringing the fork up to date. The change to `adorn_totals.R` is only in one of the commits, but I don't know how to remove the other 4 from the PR.

### CODE REVIEW

**Comment by francisbarton** [comment]
Looks like a good opportunity for me to practise writing tests!

**Comment by sfirke** [comment]
I just skimmed it all and it looks good!  Thank you.  I will find some time in the next few days to review and run line by line, then make any edits if needed and merge.  Only thing left I see immediately is to add this update to NEWS.Md, thanking you for the contribution - feel free to add that if you like :grin:

File: R/adorn_totals.R [hunk 1]
```diff
@@ -133,9 +144,28 @@ adorn_totals <- function(dat, where = "row", fill = "-", na.rm = TRUE, name = "T
...
+    #     dplyr::bind_cols(dat, .)
+    # }
+    
+    # dplyr >= 1.0 version:
+    # if ("col" %in% where) {
+    #   # Add totals col
+    #   dat <- dat %>%
+    #     dplyr::rowwise() %>% 
+    #     dplyr::mutate(name[2] = sum(c_across(all_of(cols_to_total) & where(is.numeric)), na.rm = na.rm)) %>% 
+    #     dplyr::ungroup()
```
Comment (sfirke) [inline R/adorn_totals.R:166]: Thanks for these!  I tried the dplyr >= 1.0 one for a while but had failing tests so just for the sake of time I kept what I had.  I got as far as changing your line 165 to `mutate("{name[2]}" := sum(c_across` - the `mutate(name[2]` wasn't working for me.

Comment (sfirke) [reply R/adorn_totals.R:166]: No need to work on it, I think status quo is okay - just sharing my experience in case we come back to this someday.

**Comment by sfirke** [comment]
hey @francisbarton I am having Git trouble and ended up with #414, a separate pull request.  Still has your commits though!  I will probably merge that one and close this one.

**Pull request merged**